composer.json

```diff
@@ -10,6 +10,7 @@
 		"issues": "https://core.trac.wordpress.org/"
 	},
 	"require": {
+		"ext-json": "*",
```

Comment:
Just because the IDE suggests adding this doesn't mean it's required :-)

I'd suggest removing this again for now and let us handle this separately in https://core.trac.wordpress.org/ticket/48086

Reply:
But that other ticket has been stale for 4 years, so why not include it as part of a commit that explicitly requires `ext-json`?

Reply:
> Correct. But the ext-json is required, though, as there is no polyfill, right?

After browsing through the discussion and comments like https://core.trac.wordpress.org/ticket/23912#comment:30, the consensus was that WP can still work somewhat without many of the extensions and they should only be put in `suggest`.

> so why not include it as part of a commit that explicitly requires ext-json?

It's not the first commit that makes use of `json_encode()`, so why now? My initial thought was that it would be more suitable to commit this change separately as part of the corresponding 48086 ticket, keeping everything in one place.

But then again, I don't have a strong opinion on this. This just affects development environments anyway.

---

Just saw that with PHP 8.0+ the JSON extension cannot be disabled anymore, so listing it in composer.json after that is superfluous. TIL!

Reply:
> the consensus was that WP can still work somewhat without many of the extensions and they should only be put in `suggest`.

The JSON extension is listed as `required` in the official docs:

- https://make.wordpress.org/hosting/handbook/server-environment/#php-extensions

@javiercasares can you confirm that?

---

I don't have a strong opinion about whether to include this change in this patch or in a separate one, though.

Reply:
Yes, after PHP8+, JSON is bundled in the PHP core, so is not anymore an extension…

Currently, the only needed extension is the mysql(i|nd) one.

AFAIK, with PHP8+ there is no way to disable JSON functions.

Reply:
> Currently, the only needed extension is the mysql(i|nd) one

Is the handbook incorrect, then?

- https://make.wordpress.org/hosting/handbook/server-environment/#php-extensions

<img width="600" alt="Screenshot 2024-02-07 at 10 12 14" src="https://github.com/WordPress/wordpress-develop/assets/3305402/c85de08b-2e4c-4649-8383-1cc08b9f26ba">

Reply:
No, the handbook is correct, because JSON was needed before PHP 8.x, that's why is required (for hosters using PHP 5 ot PHP 7).

But that's why it says that is bundled with PHP 8+. So, if you are using PHP 7.x you need to install the extension (I think in some moment Core changed something that needed JSON for some functions).

Technically, it should happen with MySQL because mysqli is bundled with PHP, but should be better to use mysqlnd.

Technically, if you install "php" (8+) without extensions, WordPress should work.

Reply:
Server requirements for contributing to WordPress and running WordPress are two separate things. This `composer.json` is for the former, the hosting handbook for the latter.

I suppose it's ok if we just leave this in, at least I was able to provide some additional context

Reply:
> No, the handbook is correct, because JSON was needed before PHP 8.x, that's why is required (for hosters using PHP 5 ot PHP 7).

Understood. Thank you, Javi!

> I suppose it's ok if we just leave this in, at least I was able to provide some additional context

Ok, resolving this now, then 🙂

Feel free to unresolve it if you change your mind.